runtime/runtime-params-estimator/estimator-warehouse/src/db.rs

```diff
@@ -191,3 +191,280 @@ impl Metric {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::db::{Db, EstimationRow};
+    use crate::Metric;
+    use chrono::NaiveDateTime;
+    use rusqlite::params;
+
+    #[track_caller]
+    fn check_get_single(db: &Db, name: &str, commit: &str, metric: Metric, expected_gas: f64) {
+        let row = &EstimationRow::get(db, name, commit, metric).unwrap()[0];
+        assert_eq!(row.name, name);
+        assert_eq!(row.commit_hash, commit);
+        assert!(row.is_metric(metric));
+        assert_eq!(row.gas, expected_gas);
+    }
```

Comment:
Here, I'd probably *not* test this fn in isolation, and instead rely on checks done via the top-level `check` command. The litmus test here is "would test make sense if we change the impl?". Here, it doesn't seem to be the case -- i can imagine a future impl which dispenses with loading a single row and always does bulk loading.